pkg/router/template/router.go

```diff
@@ -222,18 +223,66 @@ func (r *templateRouter) EnableRateLimiter(interval int, handlerFunc ratelimiter
 	glog.V(2).Infof("Template router will coalesce reloads within %v seconds of each other", interval)
 }
 
-// writeDefaultCert is called a single time during init to write out the default certificate
+// secretToPem composes a PEM file at the output directory from an input private key and crt file.
+func secretToPem(outName string) error {
+	// The secret, when present, is mounted on /etc/pki/tls/private
+	// The secret has two components crt.tls and key.tls
+	// When the default cert is provided by the admin it is a pem
+	//   tls.crt is the supplied pem and tls.key is the key
+	//   extracted from the pem
+	// When the admin does not provide a default cert, the secret
+	//   is created via the service annotation. In this case
+	//   tls.crt is the cert and tls.key is the key
+	//   The crt and key are concatenated to form the needed pem
+
+	certDir := os.Getenv("DEFAULT_CERTIFICATE_DIR")
+	var fileCrtName = filepath.Join(certDir, "tls.crt")
+	var fileKeyName = filepath.Join(certDir, "tls.key")
+	pemBlock, err := ioutil.ReadFile(fileCrtName)
+	if err != nil {
+		return err
+	}
+	keys, err := cmdutil.PrivateKeysFromPEM(pemBlock)
+	if err != nil {
+		return err
+	}
+	if len(keys) == 0 {
+		// Try to get the key from the tls.key file
+		keyBlock, err := ioutil.ReadFile(fileKeyName)
+		if err != nil {
+			return err
+		}
+		pemBlock = append(pemBlock, keyBlock...)
+	}
+	return ioutil.WriteFile(outName, pemBlock, 0444)
+}
+
+// writeDefaultCert ensures that the default certificate in pem format is in a file
+// and the file name is set in r.defaultCertificatePath
 func (r *templateRouter) writeDefaultCert() error {
+	// A path is provided so nothing needs to be done
+	if len(r.defaultCertificatePath) != 0 {
```

Comment:
This is a change of behavior... before, I think the DEFAULT_CERTIFICATE trumped the DEFAULT_CERTIFICATE_PATH.  This switches that.